osu.Game/OsuGameBase.cs

```diff
@@ -626,15 +629,25 @@ private void onRulesetChanged(ValueChangedEvent<RulesetInfo> r)
                 return;
             }
 
-            var previouslySelectedMods = SelectedMods.Value.ToArray();
+            //for some reason emptying SelectedMods resets all SettingSources Bindables to default value
+            var previouslySelectedMods = SelectedMods.Value.Select(mod => mod.DeepClone()).ToArray();
```

Comment:
going to need a better explanation than "for some reason"

Reply:
Honestly I have no idea why it is behaving that way, but doesn't seem to be intended. Might just remove the "for some reason" and leave it at that

Reply:
Upon further discussion I arrived at the realization that the `Mod` instances reset their settings on being deselected. In the end I managed to get rid of `DeepClone()`, which also resulted in the removal of mod select/deselect samples being played. Ideally we would want to play the mod deselect sample on mods, which didn't get converted.